#!/usr/bin/env python
PACKAGE = "op2_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x_offset",    double_t,    0, "x_offset", -10.0000,  -100, 100)
gen.add("y_offset",    double_t,    0, "y_offset", -5,  -100, 100)
gen.add("z_offset",    double_t,    0, "z_offset", 20,  -100, 100)
gen.add("roll_offset",    double_t,    0, "roll_offset", 0.0000,  -100, 100)
gen.add("pitch_offset",    double_t,    0, "pitch_offset", 0,  -100, 100)
gen.add("yaw_offset",    double_t,    0, "yaw_offset", 0,  -100, 100)
gen.add("hip_pitch_offset",    double_t,    0, "hip_pitch_offset", 13,  -100, 100)
gen.add("period_time",    int_t,    0, "period_time", 1000,  0, 10000)
gen.add("dsp_ratio",    double_t,    0, "dsp_ratio", 0.5,  0, 1)
gen.add("step_forward_back_ratio",    double_t,    0, "step_forward_back_ratio", .28,  0, 1)
gen.add("foot_height",    double_t,    0, "foot_height", 40,  0, 100)
gen.add("swing_right_left",    double_t,    0, "swing_right_left", 30,  0, 100)
gen.add("swing_top_down",    int_t,    0, "swing_top_down", 5,  0, 100)
gen.add("pelvis_offset",    double_t,    0, "pelvis_offset", .3,  0, 10)
gen.add("balance_ankle_pitch_gain",    double_t,    0, "balance_ankle_pitch_gain", .5,  0, 10)
gen.add("balance_ankle_roll_gain",    double_t,    0, "An Integer parameter", 10,  0, 10)
gen.add("p_gain",    int_t,    0, "p_gain", 32,  0, 128)
gen.add("i_gain",    int_t,    0, "i_gain", 0,  0, 128)
gen.add("d_gain",    int_t,    0, "d_gain", 0,  0, 128)

exit(gen.generate(PACKAGE, "op2_controllers", "Walking"))